beacon_node/lighthouse_network/src/discovery/subnet_predicate.rs

```diff
@@ -30,23 +28,23 @@ where
         let sync_committee_bitfield: Result<EnrSyncCommitteeBitfield<E>, _> =
             enr.sync_committee_bitfield::<E>();
 
-        let custody_subnet_count = enr.custody_subnet_count::<E>(&spec_clone);
-
         let predicate = subnets.iter().any(|subnet| match subnet {
             Subnet::Attestation(s) => attestation_bitfield
                 .get(*s.deref() as usize)
                 .unwrap_or(false),
             Subnet::SyncCommittee(s) => sync_committee_bitfield
                 .as_ref()
                 .map_or(false, |b| b.get(*s.deref() as usize).unwrap_or(false)),
-            Subnet::DataColumn(s) => {
-                let mut subnets = DataColumnSubnetId::compute_custody_subnets::<E>(
-                    enr.node_id().raw().into(),
-                    custody_subnet_count,
-                    &spec,
-                );
-                subnets.contains(s)
-            }
+            // If we're unable to get custody subnet count from ENR, fallback to
+            // `custody_requirement`.
+            // If we're unable to compute custody subnets given a custody subnet count, return `false`.
+            Subnet::DataColumn(s) => DataColumnSubnetId::compute_custody_subnets::<E>(
+                enr.node_id().raw().into(),
+                enr.custody_subnet_count::<E>()
+                    .unwrap_or_else(|_| spec.custody_requirement),
+                &spec,
+            )
+            .map_or(false, |mut subnets| subnets.contains(s)),
```

Comment:
I think it's fine to default to `false` here - it's consistent behaviour with the attestation and sync committee bitfield, and it filters out potential bad peers during discovery.

Reply:
outdated